instrumentation/cassandra/cassandra-4.0/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_0/CassandraSingletons.java

```diff
@@ -32,8 +31,6 @@ public final class CassandraSingletons {
             .addAttributesExtractor(
                 SqlClientAttributesExtractor.builder(attributesGetter)
                     .setTableAttribute(SemanticAttributes.DB_CASSANDRA_TABLE)
-                    .setStatementSanitizationEnabled(
```

Comment:
This is just to have the build phase successfully passing, to avoid:

> package io.opentelemetry.javaagent.bootstrap.internal does not exist import io.opentelemetry.javaagent.bootstrap.internal.CommonConfig;

Currently I don't have a better solution than create a full copy of the `TracingCqlSession.java` (this class is used here) in the `cassandra.v4_14` module and move the "old" `TracingCqlSession.java` from `library` module to `javaagent` module (together with `CassandraSingletons`)

Reply:
That error happens because the library modules cannot access javaagent code.
Instead of just copying the `CassandraSingletons` class, could you add a `CassandraTelemetry` & `CassandraTelemetryBuilder` classes that allow programatic configuration? You can take a look at the mongodb [telemetry](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/mongo/mongo-3.1/library/src/main/java/io/opentelemetry/instrumentation/mongo/v3_1/MongoTelemetry.java) and [builder](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/mongo/mongo-3.1/library/src/main/java/io/opentelemetry/instrumentation/mongo/v3_1/MongoTelemetryBuilder.java) classes for an example.

Reply:
I'm not seeing the solution here. It's not just a problem related to `CassandraSingletons`... `CassandraSingletons` contains [jaavagent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4cbfd7da8b80d4f76e56537279f1f15a012ad51a/instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraSingletons.java#L15) code. If we remove this dependency by adding (for example) a boolean parameter for the [instrumenter()](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4cbfd7da8b80d4f76e56537279f1f15a012ad51a/instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraSingletons.java#L44) method there is still the problem that we need to pass this parameter (this `CommonConfig.get().isStatementSanitizationEnabled()`) in the `TracingCqlSession` class every time we call `instrumenter()`.

However this `TracingCqlSession` is part of the `library` module (i.e. we cannot have this code `instrumenter(CommonConfig.get().isStatementSanitizationEnabled())`) otherwise we still have the same problem.

We could add a boolean paramenter in the `TracingCqlSession` and finally configure this boolean flag in the [CompletionStageFunction](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4cbfd7da8b80d4f76e56537279f1f15a012ad51a/instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CompletionStageFunction.java#L22) (it's javaagent code): in other words pass the configuration in the TracingCqlSession constructor. Otherwise we still need to move `TracingCqlSession` in the javaagent module.

Do you agree or I didn't get your suggestion?

Reply:
What I meant was that `CassandraSingletons` should not be present in the library instrumentation module at all. The `*Singletons` classes are (by convention) only used in javaagent instrumentations, where you need to have some sort of a global var/instance.

Consider the following example/pseudocode:
```java
class CassandraTelemetry {
  static CassandraTelemetryBuilder(OpenTelemetry otel) {
    return new CassandraTelemetryBuilder(otel);
  }
  
  private final Instrumenter<CassandraRequest, ExecutionInfo> instrumenter;
  private final boolean statementSanitizationEnabled;
  // constructor

  public CqlSession decorate(CqlSession original) {
    return new TracingCqlSession(instrumenter, statementSanitizationEnabled, original);
  }
}

class CassandraTelemetryBuilder {
  CassandraTelemetryBuilder setStatementSanitizationEnabled(boolean enabled) {
    // ...
  }
  
  CassandraTelemetry build() {
    // construct the Instrumenter
    return new CassandraTelemetry(instrumenter, statementSanitizationEnabled);
  }
}
```

You're not using `CommonConfig` anywhere, you're allowing the user of `CassandraTelemetryBuilder` to pass the option programmatically through a builder method. The javaagent `CassandraSingletons` class could then look just like that:

```java
class CassandraSingletons {
  static final CassandraTelemetry telemetry = CassandraTelemetry.builder(GlobalOpenTelemetry.get())
      .setStatementSanitizationEnabled(CommonConfig....)
      .build();
}
```

Reply:
Thanks, it is clearer now. I [committed](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6441/commits/128f0f620b863a774d02ded395bd8067ff2e6030) this changes.